core/server/i18n.js

```diff
@@ -94,23 +121,88 @@ I18n = {
      *  - Load proper language file in to memory
      *  - Polyfill node.js if it does not have Intl support or support for a particular locale
      */
-    init: function init() {
-        // read file for current locale and keep its content in memory
-        blos = fs.readFileSync(__dirname + '/translations/' + currentLocale + '.json');
-
-        // if translation file is not valid, you will see an error
-        try {
-            blos = JSON.parse(blos);
-        } catch (err) {
-            blos = undefined;
-            throw err;
+    init: function init(i18nCore, i18nTheme) {
+        // This function is called during Ghost's initialization, when switching language or theme...
+        // For the first initialization call, settings for language and theme are not yet available,
+        // and English default is used for core; settings are available shortly after, allowing
+        // full internationalization of core and theme.
+
+        var blosTheme, hasBuiltInLocaleData, IntlPolyfill, backendTranslations;
+
+        if (blos === undefined) {
+            // If not in memory, load translations for core and theme.
+            i18nCore = true;
+            i18nTheme = true;
+        }
+
+        // Switch to enable/disable translations of back-end messages from core .js files.
+        // - true: enabled; optional files like content/locales/es.json, etc. are used when available.
+        // - false: disabled; only the default English file core/server/translations/en.json is used.
+        // Before enabling back-end message translations, see previous tasks on issue #6526
+        // (error codes or identifiers, error message translation at the point of display...)
+        backendTranslations = false;
+
+        currentLocale = settingsCache.get('default_locale') || 'en';
+        activeTheme = settingsCache.get('active_theme') || '';
+
+        // Reading files for current locale and active theme and keeping their content in memory
+        // (during Ghost's initialization, when English is the language in settings, the second call
+        // to i18n.init() doesn't read again the default English file for core if already in memory).
+        if (i18nCore) {
+            // Reading translation file for core .js files.
+            if (currentLocale === 'en' || !backendTranslations) {
+                blos = fs.readFileSync(path.join(__dirname, 'translations', 'en.json'));
+            } else {
+                // Preventing missing file.
+                try {
+                    blos = fs.readFileSync(path.join(__dirname, '..', '..', 'content', 'locales', currentLocale + '.json'));
```

Comment:
We don't support any core translations for the first version, [see](https://github.com/TryGhost/Ghost/pull/8437#issuecomment-330309612).

Reply:
Yes, that's why back-end translations (of server-side messages from core _.js_ files) are currently disabled.

Back-end translations were already almost completely developed in Ghost before my i18n work here. Most server-side messages are in fact translations to English using `i18n.t()`. So, in May, I simply completed its development as an optional feature, in addition to the new front-end translations.

When, in September, the team _"decided against Ghost Core translations for the first iteration"_, I added a switch to enable/disable them, because back-end translations were already developed and tested. In this way, in a future version, for example if error numbers or other features are added to the core translation files, then `I18n.backendTranslations()` can just be set to `true` instead of the current `false` (which uses the default English), with no need to develop core translations again, and they will work.